### PR TITLE
Migration: Add migration experiment link

### PR DESCRIPTION
**Pull request by gabrielcaires**
Related to #93881

## Proposed Changes
* Integrate the link to the new migration based on the experiment config
* Shows a "Redirecting" message while the experiment is not fully loaded and the user tries to click.  ( it is hard to reproduce, so automated tests are testing ).

## Testing Instructions

Scenario 1: Going to the new behavior 
* Go to `/site`
* Click on `Add new site` > Import an existing site
* Check if you were redirected to `/setup/migration`

Scenario 2: Old behavior
* Go to `/site`
* Using the Developer Tools check the local storage already have the key   `explat-experiment--calypso_optimized_migration_flow` 
* The value will contain a JSON;
* Update the  `variationName` field to `"treatment"` instead of null; 
* Reload the page
* Click on `Add new site` > Import an existing site
* Check if you were redirected to `/setup/hosted-site-migration`

NOTE: Maybe you will need to repeat the local storage steps sometimes because there is a logic on the exPlat framework recreating it. 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/hosting/sites/hooks/use-sites-dashboard-import-site-url.ts
Comment (renatho) [inline client/hosting/sites/hooks/use-sites-dashboard-import-site-url.ts:9]: Thinking more about this, I wonder if this approach is correct. 🤔 Maybe @daledupreez could help us to understand if it's correct or not.

Basically, we're trying to make the experiment redirecting users from this link to `/setup/hosted-site-migration` or to `/setup/migration`.

But we're initializing the experiment before the user clicks on the link. In theory, all the users clicking on this link could be part of the "control" bucket, so we wouldn't have any user from the "treatment" in our experiment. Is it fine since statistically it shouldn't happen? Or should we apply the experiment only after the user clicks on the link, making sure that we'll distribute 50/50?

My same question would apply to the `/move` and `/hosting`, where we plan to have an experiment changing the `href` of the links. In this case, we plan to use the Landpack experiment block. In this case, if we plan to apply the experiment only after the click, I think it would be much more complex, probably needing to use another approach. 🤔

File: client/hosting/sites/hooks/use-sites-dashboard-import-site-url.ts
Comment (renatho) [inline client/hosting/sites/hooks/use-sites-dashboard-import-site-url.ts]: Ah, sorry! It came from my commit. I wrote it in the wrong way.
The treatment should be the `/setup/migration`. Could you fix this before the merge?

**Review by renatho — APPROVED** [review]
It looks good and works well!
We just need to reverse the conditional and, as we talked, we can tweak later if something is needed for https://github.com/Automattic/wp-calypso/pull/93984/files#r1734831351.

File: client/hosting/sites/hooks/use-sites-dashboard-import-site-url.ts
Comment (gabrielcaires) [reply client/hosting/sites/hooks/use-sites-dashboard-import-site-url.ts]: Fixe [02c3093](https://github.com/Automattic/wp-calypso/pull/93984/commits/02c3093d780fe7d20fb845d19a37699b1eec6e13)

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16617437

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @gabrielcaires, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Redirecting`

Thank you in advance!

File: client/hosting/sites/hooks/use-sites-dashboard-import-site-url.ts
Comment (necronet) [reply client/hosting/sites/hooks/use-sites-dashboard-import-site-url.ts:9]: What would happen if I get to either of those links directly, since the assignment is happening here and the link generation depends on the experiment assignment I'd think that the this will be a source of crossover ratios. cc @aaronyan (see internal FG page `the-experimentation-platform/experiment-health/#ratios`)

> Or should we apply the experiment only after the user clicks on the link, making sure that we'll distribute 50/50

This sounds like the most natural from an experiment assignment perspective, but if you're use case calls for having different urls the assignment code here is implemented correctly.

Comment (renatho) [reply client/hosting/sites/hooks/use-sites-dashboard-import-site-url.ts:9]: Thank you, @necronet! I'll wait for @aaronyan's answer to see if it's really an issue and if we should change anything.

Comment (aaronyan) [reply client/hosting/sites/hooks/use-sites-dashboard-import-site-url.ts:9]: Preloading the experiment assignment before the divergence in experience is okay and nothing to worry about. Variation assignment is deterministic, meaning that provided the same `experimet_id` + `user_id`, you will always receive the same assigned variation.

We sometimes preload assignment to make sure there is no delay in the experiment. And then we use the exposure event (visiting `control` experience or `treatment` experience) to filter out users that were "exposed" to the experiment. Exposures help clean up the experiment data and reduces noise in the analysis.

I don't see any issues here!

Comment (aaronyan) [reply client/hosting/sites/hooks/use-sites-dashboard-import-site-url.ts:9]: > links directly

If there are ways for users to be exposed to multiple experiences, there may be cause for concern. But generally, most users won't know what direct URL to enter to get into the other experience, so it's usually not an issue.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16617437).

**Pull request merged**
